### PR TITLE
deprecate pynio backend

### LINKED ISSUE #4491: deprecate pynio backend
We are currently not testing with the newest version of netCDF4 because it is incompatible with pynio (the newest version is [1.5.4](https://github.com/Unidata/netcdf4-python/blob/master/Changelog), we are at [1.5.3](https://dev.azure.com/xarray/xarray/_build/results?buildId=3974&view=logs&j=ba13898e-1dfb-5ace-9966-8b7af3677790&t=0a8d5551-2e87-575f-4689-3e85d9688898&l=220)). This is unlikely to be fixed, see conda-forge/pynio-feedstock#90.

Therefore we need to think how to setup the tests so we use the newest version of netCDF4. Maybe just remove it from `py38.yml`?

And long term what to do with the pynio backend? Deprecate? Move to an external repo?

### PR DESCRIPTION
**Pull request by jhamman**
This PR finally deprecates the PyNIO backend. PyNIO is technically in maintenance mode but it hasn't had any maintenance in 4+ years. Its conda packages cannot be installed in any of our test environments. I have added a future warning to the `NioDataStore.__init__` method and noted the deprecation in the IO docs.

- [x] Closes #4491 
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/backends/pynio_.py [hunk 1]
```diff
@@ -55,6 +57,12 @@ class NioDataStore(AbstractDataStore):
     def __init__(self, filename, mode="r", lock=None, **kwargs):
         import Nio
 
+        warnings.warn(
+            "The PyNIO backend is Deprecated and will be removed from Xarray in a future release. "
+            "See https://github.com/pydata/xarray/issues/4491 for more information",
+            FutureWarning,
```
Comment (headtr1ck) [inline xarray/backends/pynio_.py:63]: Maybe we can directly use a `DeprecationWarning`.

**Review by headtr1ck — APPROVED** [review]

**Comment by headtr1ck** [comment]
Maybe we should remove the pynio backend tests alltogether since they will be skipped anyway if no env has pynio enabled anymore?

**Comment by jhamman** [comment]
> Maybe we should remove the pynio backend tests all together since they will be skipped anyway if no env has pynio enabled anymore?

I think we should keep them in the test suite as long as the backend is in the code. Although we able to install pynio in our test envs, I suspect it is still possible to construct an environment centered around the pynio that would still work.

**Comment by Illviljan** [comment]
Thanks, @jhamman !

**Pull request merged**